src/NHibernate.Test/NHSpecificTest/GH1413/FixtureByCode.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Linq;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Mapping.ByCode;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1413
+{
+	[TestFixture]
+	public class ByCodeFixture : TestCaseMappingByCode
+	{
+		private Guid ParentId;
+
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<EntityParent>(rc =>
+			{
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.Name);
+				rc.ManyToOne(ep => ep.Child);
+				rc.Bag(x => x.Children, m =>
+				{
+					m.Cascade(Mapping.ByCode.Cascade.All);
+					m.Inverse(true);
+				}, a => a.OneToMany()
+				);
+			});
+
+			mapper.Class<EntityChild>(rc =>
+			{
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.Name);
+			});
+
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		protected override void OnSetUp()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				var child = new EntityChild { Name = "InitialChild" };
+				var parent = new EntityParent
+				{
+					Name = "InitialParent",
+					Child = child
+				};
+				session.Save(child);
+				session.Save(parent);
+				
+				session.Flush();
+				transaction.Commit();
+				ParentId = parent.Id;
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public void SessionIsDirtyShouldNotFailForNewManyToOneObject()
+		{
+			using (ISession session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				EntityParent parent = GetParent(session);
+
+				//parent.Child entity is not cascaded, I want to save it explictilty later
+				parent.Child = new EntityChild { Name = "NewManyToOneChild" };
+
+				bool isDirty = false;
+				Assert.That(() => isDirty = session.IsDirty(), Throws.Nothing, () => "ISession.IsDirty() call should not fail for transient  many-to-one object referenced in session.");
```

Comment:
I was not knowing it could take a delegate for the error message (third argument). A string argument is enough for such cases indeed. (That is just a small detail, not important.)